Objects/floatobject.c

```diff
@@ -131,11 +131,7 @@ PyFloat_FromDouble(double fval)
 #ifdef WITH_FREELISTS
     struct _Py_float_state *state = get_float_state();
     op = state->free_list;
-    if (op != NULL) {
-#ifdef Py_DEBUG
-        // PyFloat_FromDouble() must not be called after _PyFloat_Fini()
-        assert(state->numfree != -1);
-#endif
+    if (op != NULL && state->numfree > 0) {
```

Comment:
I don't think we need extra checks when allocating from the free-lists. The existing checks should be sufficient.